### PR TITLE
Initial commit for MySQL support

### PR DESCRIPTION
**Pull request by tonedef**
Signed-off-by: Alex Kobylk <akobylk@linkedin.com>

## Hey, I just made a Pull Request!
Added support for MySQL for plugins/catalog-backend. This is a first pass. There is more to do, such as ensure metadata does not exceed MySQL row length limits.

#### :heavy_check_mark: Checklist
_yarn test_  passes locally with MySQL and with sqlite

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/catalog-backend/src/service/standaloneServer.ts [hunk 1]
```diff
@@ -47,7 +47,15 @@ export async function startStandaloneServer(
     const manager = DatabaseManager.fromConfig(
       new ConfigReader({
         backend: {
-          database: { client: 'better-sqlite3', connection: ':memory:' },
+          database: {
+            client: 'mysql2',
```
Comment (jhaals) [inline plugins/catalog-backend/src/service/standaloneServer.ts:51]: I believe this was used for local testing. Should be reverted :)

**Review by jhaals — COMMENTED** [review]
Thanks for taking a stab at this, looks like this currently break migrations when running postgres as the database.

File: plugins/catalog-backend/src/service/standaloneServer.ts (hunk 1 above)
Comment (tonedef) [reply plugins/catalog-backend/src/service/standaloneServer.ts:51]: ah, yes, it was. Thanks for point it out. I'll change it.

**Comment by tonedef** [comment]
It looks like this is passing. Although previously it was failing on CI/verify(14.x). Anyway to run that locally? E.g. a docker image etc?

**Review by Rugvip — COMMENTED** [review]
@tonedef You'll need to run the tests towards PostgreSQL and SQLite as well, that's likely were the failures are right?

Thank you for taking a stab at this btw! It gives us a really good idea of what it would mean to support MySQL as well, since the catalog is likely the most complex user of the DB so far.

We'll have to weight the maintenance burden vs benefit of supporting MySQL carefully, so it's very helpful to se the exact pieces of code that are needed to make it work. We will need to have all code be able to work towards all 3 DBMSs at once though, since that is in the end what we will need.

**Comment by tonedef** [comment]
Thanks @Rugvip! Appreciate the feedback. I'll def. make sure these pass. I was a bit blocked with CI testing locally, I just need to setup  docker containers that match your CI env. so I can debug, but with @freben's help, I think I have a path forward with that.

Even if MySQL is not officially supported, getting the catalog working would be a great first step. We'll make it happen.

**Comment by benjdlambert** [comment]
Gentle ping here @tonedef

**Pull request closed without merging**